src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+            yield return new object[] { 10000, TestType.String, 1024, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 128, serializerType };
+            yield return new object[] { 1000, TestType.ListOfInt, 1024, serializerType };
+            yield return new object[] { 1, TestType.ListOfInt, 1024 * 1024, serializerType };
+            yield return new object[] { 1000, TestType.Dictionary, 128, serializerType };
+            yield return new object[] { 100, TestType.Dictionary, 1024, serializerType };
+            yield return new object[] { 10, TestType.SimpleType, 1, serializerType };
+            yield return new object[] { 1, TestType.SimpleType, 15, serializerType };
+            yield return new object[] { 10000, TestType.ISerializable, -1, serializerType };
+            yield return new object[] { 10000, TestType.XmlElement, -1, serializerType };
+        }
+
+        #endregion
+
+        #region Methods to create object for serialization tests
+
+        public static SimpleTypeWihtMoreProperties CreateObject(int height, int parentId, int currentId, int collectionSize, int childListSize)
+        {
+            int index = parentId * childListSize + (currentId + 1);
+            var obj = new SimpleTypeWihtMoreProperties()
+            {
+                IntProperty = index,
+                StringProperty = index + " string value",
+                EnumProperty = (MyEnum)(index % (Enum.GetNames(typeof(MyEnum)).Length)),
+                CollectionProperty = new List<string>(),
+                SimpleTypeList = new List<SimpleTypeWihtMoreProperties>()
+            };
+            for (int i = 0; i < collectionSize; ++i)
+            {
+                obj.CollectionProperty.Add(index + "." + i);
+            }
+            if (height > 1)
+            {
+                for (int i = 0; i < childListSize; ++i)
+                {
+                    obj.SimpleTypeList.Add(CreateObject(height - 1, index, i, collectionSize, childListSize));
+                }
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>();
+            for (int i = 0; i < count; ++i)
+            {
+                dictOfIntString[i] = i.ToString();
+            }
+
+            return dictOfIntString;
+        }
+
+        public static List<int> CreateListOfInt(int count)
+        {
+            return Enumerable.Range(0, count).ToList();
+        }
+
+        public static object GetSerializationObject(TestType testType, int testSize)
```

Comment:
I don't see much value of having TestType, it would only add extra maintenance cost. I think we should encapsulate all information related to a certain test into a separate type, that way we can simply adding new types for new tests.